File: languages/java/src/main/java/com/bitwarden/sdk/AuthClient.java
package com.bitwarden.sdk;

import com.bitwarden.sdk.schema.*;

import java.util.function.Function;

public class AuthClient {

    private final CommandRunner commandRunner;

    AuthClient(CommandRunner commandRunner) {
        this.commandRunner = commandRunner;
    }

    public APIKeyLoginResponse loginAccessToken(String accessToken, String stateFile) {
        Command command = new Command();
        AccessTokenLoginRequest accessTokenLoginRequest = new AccessTokenLoginRequest();
        accessTokenLoginRequest.setAccessToken(accessToken);
        accessTokenLoginRequest.setStateFile(stateFile);

        command.setLoginAccessToken(accessTokenLoginRequest);

        ResponseForAPIKeyLoginResponse response = commandRunner.runCommand(command,
            BitwardenClient.throwingFunctionWrapper(Converter::ResponseForAPIKeyLoginResponseFromJsonString));

        if (response == null || !response.getSuccess()) {
            throw new BitwardenClientException(response != null ? response.getErrorMessage() : "Login failed");
        }

        return response.getData();
    }
}

File: languages/csharp/Bitwarden.Sdk/AuthClient.cs
namespace Bitwarden.Sdk;

public class AuthClient
{
    private readonly CommandRunner _commandRunner;

    internal AuthClient(CommandRunner commandRunner)
    {
        _commandRunner = commandRunner;
    }

    public async Task LoginAccessTokenAsync(string accessToken, string stateFile = "", CancellationToken cancellationToken = default)
    {
        var command = new Command { LoginAccessToken = new AccessTokenLoginRequest { AccessToken = accessToken, StateFile = stateFile } };
        var response = await _commandRunner.RunCommandAsync<ResponseForApiKeyLoginResponse>(command, cancellationToken);
        if (response is not { Success: true })
        {
            throw new BitwardenAuthException(response != null ? response.ErrorMessage : "Login failed");
        }
    }
}
